src/main/java/types/DisplayType.java

```diff
@@ -49,5 +50,49 @@ public enum DisplayType {
      * @date 23 Apr 2017
      * @status added
      */
-    SPICE;
+    @Deprecated
+    SPICE,
+
+    /**
+     * Display of type CIRRUS
+     * @author Jean-Louis Dupond <jean-louis@dupond.be>
+     * @date 26 May 2023
+     * @status added
+     */
+    @Deprecated
+    CIRRUS,
```

Comment:
you're mixing video devices and graphic devices here - we intentionally didn't add the video devices to the API because they are represented differently

Reply:
Back in the early days, oVirt only had DisplayType.
But then the GraphicsType was added: https://github.com/oVirt/ovirt-engine/commit/9cdd6f887ee73fa36d9dd1f7c898fd30fce4f835

So now oVirt Engine itself has both types:

The DisplayType (QXL/VGA/BOCHS/etc)
https://github.com/oVirt/ovirt-engine/blob/master/backend/manager/modules/common/src/main/java/org/ovirt/engine/core/common/businessentities/DisplayType.java

And the GraphicsType (VNC/SPICE)
https://github.com/oVirt/ovirt-engine/blob/master/backend/manager/modules/common/src/main/java/org/ovirt/engine/core/common/businessentities/GraphicsType.java


But for some reason, you can't set the real DisplayType from the REST API.
While you can just select it from the UI.

So we need to add the real DisplayTypes to the model, so we can then allow the REST API code to set the DisplayType.
Cause without this, you can't set your VM to VGA-VNC for example, as it will always prefer QXL if you pass DisplayType.VNC to the API.

Reply:
> So we need to add the real DisplayTypes to the model, so we can then allow the REST API code to set the DisplayType. Cause without this, you can't set your VM to VGA-VNC for example, as it will always prefer QXL if you pass DisplayType.VNC to the API.

can you please elaborate on that? it should have been handled by [making VGA the default video type](https://github.com/oVirt/ovirt-engine/pull/75) - what cluster version do you use? how the blank template looks like in terms of video and graphic devices? and how do you provision the vm (from scratch, i.e., from the blank template or from an existing template - in case of the latter, how that template is set in terms of video and graphic devices?)

Reply:
> > So we need to add the real DisplayTypes to the model, so we can then allow the REST API code to set the DisplayType. Cause without this, you can't set your VM to VGA-VNC for example, as it will always prefer QXL if you pass DisplayType.VNC to the API.
> 
> can you please elaborate on that? it should have been handled by [making VGA the default video type](https://github.com/oVirt/ovirt-engine/pull/75)

ah no, that's not the right one - @liranr23 do you remember how we managed to set the display type properly? was it by changing the osinfo?

Reply:
This was not about new VM's.
We had a cluster with +500 VM's on CentOS 8 hypervisors. All QXL - VNC/SPICE.

But to upgrade to CentOS 9, all VM's had to be changed to VGA - VNC (as QXL is gone in CentOS 9).

So when trying to do that, we've noticed there was no way to set this, as there is no 'VGA' DisplayType in the API. You could use DisplayType.VNC, but it preferred QXL over VGA, and didn't change the display to VGA like we needed.

Reply:
> > > So we need to add the real DisplayTypes to the model, so we can then allow the REST API code to set the DisplayType. Cause without this, you can't set your VM to VGA-VNC for example, as it will always prefer QXL if you pass DisplayType.VNC to the API.
> > 
> > 
> > can you please elaborate on that? it should have been handled by [making VGA the default video type](https://github.com/oVirt/ovirt-engine/pull/75)
> 
> ah no, that's not the right one - @liranr23 do you remember how we managed to set the display type properly? was it by changing the osinfo?

hard to recall. i think we change osinfo. but maybe we did something also on AsyncDataProvider/UnitVmModel or elsewhere, where we "preferred" something.

Reply:
https://github.com/oVirt/ovirt-engine/blob/master/backend/manager/modules/restapi/jaxrs/src/main/java/org/ovirt/engine/api/restapi/util/DisplayHelper.java#L129

Here we set the DisplayType depending on the GraphicsType. With a preference for SPICE.
See the note: https://github.com/oVirt/ovirt-engine/blob/master/backend/manager/modules/restapi/jaxrs/src/main/java/org/ovirt/engine/api/restapi/util/DisplayHelper.java#L108